crates/rsonpath-lib/src/classification.rs

```diff
@@ -113,42 +113,62 @@ where
         self.iter.get_offset() + self.block.as_ref().map_or(0, |b| b.idx)
     }
 
-    /// Move the state forward by `count` bytes.
+    /// Move the state forward to `index`.
     ///
     /// # Errors
     /// If the offset crosses block boundaries, then a new block is read from the underlying
     /// [`Input`](crate::input::Input) implementation, which can fail.
     ///
     /// # Panics
-    /// If the `count` is not positive.
+    /// If the `index` is not ahead of the current position of the state ([`get_idx`](ResumeClassifierState::get_idx)).
     #[inline]
     #[allow(clippy::panic_in_result_fn)]
-    pub fn offset_bytes(&mut self, count: isize) -> Result<(), InputError> {
-        assert!(count > 0);
-        let count = count as usize;
+    pub fn forward_to(&mut self, index: usize) -> Result<(), InputError> {
+        let current_block_start = self.iter.get_offset();
+        let current_block_idx = self.block.as_ref().map_or(0, |b| b.idx);
+        let current_idx = current_block_start + current_block_idx;
 
-        let remaining_in_block = self.block.as_ref().map_or(0, |b| b.block.len() - b.idx);
-
-        match self.block.as_mut() {
-            Some(b) if b.block.len() - b.idx > count => {
-                b.idx += count;
-            }
-            _ => {
-                let blocks_to_advance = (count - remaining_in_block) / N;
+        debug!(
+            "Calling forward_to({index}) when the inner iter offset is {} and block idx is {:?}",
+            current_block_start, current_block_idx
```

Comment:
Theres an insignificant question of uniformity.  What's preventing
`"Calling forward_to({index}) when the inner iter offset is {current_block_start} and block idx is {current_block_idx:?}"`

Reply:
Nothing, it was a more complex expression at first and then it got simplified during refactoring at some point.

Really annoying arbitrary expressions don't work yet.